File: Spring Boot/monolith/src/main/java/pl/edu/pg/benchmarking/monolith/distance/DistanceController.java
package pl.edu.pg.benchmarking.monolith.distance;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import pl.edu.pg.benchmarking.monolith.distance.dto.DistanceRequest;
import pl.edu.pg.benchmarking.monolith.distance.dto.DistanceResponse;

import java.util.List;

@RestController
@RequestMapping("api/distance")
public class DistanceController {

    private final SphericalDistanceService distanceService;

    public DistanceController(SphericalDistanceService distanceService) {
        this.distanceService = distanceService;
    }

    @PostMapping
    public ResponseEntity<DistanceResponse> calculateDistance(@RequestBody DistanceRequest request) {
        List<Point> points = DistanceRequest.dtoToEntityMapper().apply(request);
        Point point1 = points.get(0);
        Point point2 = points.get(1);

        Double distance = distanceService.calculateDistance(
                point1.getLatitude(), point1.getLongitude(),
                point2.getLatitude(), point2.getLongitude());

        return ResponseEntity.ok(DistanceResponse.entityToDtoMapper().apply(distance));
    }
}

File: Flask/microservices/distance-service/website/distance_controller.py
import website.dto.distance_request as distance_request
import website.dto.distance_response as distance_response
from website.spherical_distance_service import SphericalDistanceService
from flask import Blueprint, request, jsonify


distance_views = Blueprint('distance', __name__)
distance_service = SphericalDistanceService()


@distance_views.route('', methods=['POST'])
def calculate_distance() -> dict:
    points = distance_request.dto_to_entity_mapper(request.json)
    distances_mapped = map(lambda points_pair: distance_service.calculate_distance(points_pair), points)
    distances_list = list(distances_mapped)
    dto = distance_response.entity_to_dto_mapper(points, distances_list)
    return jsonify(dto)
